apps/spruce/src/components/DateFilter/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { DatePicker } from "@leafygreen-ui/date-picker";
+import { DateType } from "@leafygreen-ui/date-utils";
+import { Size } from "@leafygreen-ui/tokens";
+import { subDays, subYears } from "date-fns";
+import { zIndex } from "@evg-ui/lib/constants/tokens";
+import { StringMap } from "@evg-ui/lib/types/utils";
+import { isProduction } from "utils/environmentVariables";
+
+interface DateFilterProps {
+  value: string;
+  onChange: (v: string) => void;
+  dataCyProps?: StringMap;
+  showLabel?: boolean;
+  size?: Size;
+}
+
+export const DateFilter: React.FC<DateFilterProps> = ({
+  dataCyProps,
+  onChange,
+  showLabel = false,
+  size = Size.Default,
+  value,
+}) => {
+  const handleChange = (updatedValue?: DateType) => {
+    if (updatedValue) {
+      // Dates do not store timezone information; all dates should be represented in UTC.
+      const year = updatedValue.getUTCFullYear();
+      const month = updatedValue.getUTCMonth() + 1;
+      const day = updatedValue.getUTCDate();
+      const utcDate = `${year}-${month.toString().padStart(2, "0")}-${day.toString().padStart(2, "0")}`;
+      onChange(utcDate);
+    }
+  };
+
+  return (
+    <DatePicker
+      data-cy="date-picker"
+      label={showLabel ? "Go to Date" : ""}
+      max={today}
+      // Testing environments should not have a minimum date restriction due to static test data.
+      min={isProduction() ? subDays(subYears(today, 1), 1) : undefined}
```

Comment:
This makes me sad :( 
I wonder if we should use [this](https://docs.cypress.io/api/commands/clock) in the future to ensure consistent dates so our tests don't get stale

Reply:
I think we had looked at `cy.clock()` but there was some kind of problem with modals? Anyways, I will create a ticket for it

Reply:
Actually, nevermind, I do think it might be a good idea to keep it this way; or else we can't test the component on our local environments.